### PR TITLE
resolve device symbols for static builds

### PR DESCRIPTION
**Pull request by mclarsen**
This solves linking problems downstream. If we did not resolve these device symbols, then all downstream apps would have to use separable compilation and nvcc linking.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -66,6 +66,9 @@ set(ENABLE_ALL_WARNINGS OFF CACHE INTERNAL "")
 # Init BLT
 ################################
 set (CUDA_SEPARABLE_COMPILATION ON CACHE BOOL "" )
+if(NOT ENABLE_SHARED_LIBS)
+#  set (CUDA_LINK_WITH_NVCC ON CACHE BOOL "")
+endif()
```
Comment (cyrush) [inline src/CMakeLists.txt]: looks like we can drop this if stmt?

**Review by cyrush — APPROVED** [review]
Looks good! Thank for finding the narrow path

File: src/CMakeLists.txt (hunk 1 above)
Comment (mclarsen) [reply src/CMakeLists.txt]: already on it.

**Comment by mclarsen** [comment]
@cyrush I am going to have to update the hashes which could be slightly painful for some.

**Comment by cyrush** [comment]
@mclarsen no worries

**Pull request merged**